autogen.sh

```diff
@@ -166,7 +168,15 @@ export do_build_configure
 MAKE=${MAKE-make}
 
 # external packages that require autogen.sh to be run for each of them
-externals="src/pm/hydra src/mpi/romio src/openpa src/mpid/ch4/netmod/ucx/ucx src/mpid/ch4/netmod/ofi/libfabric"
+externals="src/pm/hydra src/mpi/romio src/openpa"
+
+if [ -f src/mpid/ch4/netmod/ucx/ucx/autogen.sh ] ; then
+    externals="${externals} src/mpid/ch4/netmod/ucx/ucx"
+fi
+
+if [ -f src/mpid/ch4/netmod/ofi/libfabric/autogen.sh ] ; then
+    externals="${externals} src/mpid/ch4/netmod/ofi/libfabric"
+fi
```

Comment:
In what case would these not exist?  They are checked out by default when you clone `mpich`.

Reply:
Not by default. Only if you do so manually.

I don't have any need to check out UCX as I never build it. I also don't want an embedded OFI as I use a separate one that comes from a different repository than this. I need to update my OFI far more often than once per release.

I'll be carrying along the "required" submodules (confdb, MPL, hydra), but I don't plan to keep the others.

I'm not saying that you _can't_ have these embedded. Just that it's nice to not _require_ it for no reason.

Reply:
We usually do that with options to `autogen.sh` such as `-with-bindings`.  Can you rework it to make that consistent?

Reply:
Ugh...why? This stuff is very easily autodetected. I don't even know whether to make this opt in or opt out.

For `--with-bindings`, it makes sense because the code is always there but you might want to run the scripts. For this, the code is either there or it's not.

Reply:
Requiring manual deletion of the directories seems gross and hacky to me.  The option to `autogen.sh` allows me to `git clone` the repo, but simply not build some pieces.